accepted/2020/mono-convergence/security-support-new-platforms.md

```diff
@@ -0,0 +1,90 @@
+# Security support for new .NET 5 platforms
+
+## Overview
+Currently in .NET Core, we have a solid [strategy](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/cross-platform-cryptography.md) for dealing with cryptography on Linux and OSX.  At the native level, `System.Security.Cryptography.Native` relies upon OpenSSL for Linux and `System.Security.Cryptography.Native.Apple` relies upon platform specific API’s for OSX.  
+
+With the introduction of Android, iOS, and Wasm workloads, there are implications to consider that may alter our strategy and require additional work.  This document seeks to describe the implications and what options we may have.
+
+## Security Dependencies
+For `System.Security.Cryptography.Native` (Linux):
+
+[src/libraries/Native/Unix/System.Security.Cryptography.Native] (https://github.com/dotnet/runtime/tree/master/src/libraries/Native/Unix/System.Security.Cryptography.Native)
+
+For `System.Security.Cryptography.Native.Apple` (OSX):
+
+[src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple] (https://github.com/dotnet/runtime/tree/master/src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple)
+
+### Android
+There is currently no platform level API’s that can work with Android, so any native dependency we take must be bundled with the individual application.
+
+**System.Security.Cryptography.Native**
+
+The current dependency on OpenSSL could work as there seems to be support for building on Android.  This is nice because it would seemingly ‘just fit’ with what we already have.  However, there are some things to take into consideration:
+
+* Do we fork a copy of OpenSSL and build from that?
+* What is our process for shipping security vulnerability updates?
+* Should we consider alternatives?
+  * [https://tls.mbed.org/](https://tls.mbed.org/) - only solves the TLS part
+  * Any others?
+
+**System.Net.Security.Native**
+
+For authentication scenarios, we’ve relied on libgss to do the work.  It is currently unknown if we *can* support it on Android.  Some initial [discovery](https://github.com/dotnet/runtime/issues/32680#issuecomment-599307356) has been started by a community member and would need to be expanded on / validated.
+
+To move forward, we should try to decide on the best approach.
+
+Do we:
+* Investigate libgss further and flush out all the dependencies / challenges?
+* Bring most / all functionality into a managed implementation?  What does that look like?  Who would do it?
```

Comment:
What does Xamarin do today?

Reply:
We now use SocketsHttpHandler as the default, so the same as corefx.  The previous way was a custom implementation that is pretty old.

Reply:
This isn't just about HttpClient, this is about all of System.Security.Cryptography.